src/category_theory/sites/plus.lean

```diff
@@ -0,0 +1,253 @@
+
+/-- The plus construction, associating a presheaf to any presheaf.
+See `plus` below for a functorial version.
+-/
+@[simps]
+def plus_obj : Cᵒᵖ ⥤ D :=
+{ obj := λ X, colimit (J.diagram P X.unop),
+  map := λ X Y f, colim_map (J.diagram_pullback P f.unop) ≫ colimit.pre _ _,
+  map_id' := begin
+    intros X,
+    ext S,
+    dsimp,
+    simp only [diagram_pullback_app, colimit.ι_pre,
+      ι_colim_map_assoc, category.comp_id],
+    let e := S.unop.pullback_id,
+    dsimp only [functor.op, pullback_obj],
+    erw [← colimit.w _ e.inv.op, ← category.assoc],
+    convert category.id_comp _,
+    ext I,
+    dsimp,
+    simp only [multiequalizer.lift_ι, category.id_comp, category.assoc],
+    dsimp [cover.L.map, cover.L.base],
+    cases I,
+    congr,
+    simp,
+  end,
+  map_comp' := begin
+    intros X Y Z f g,
+    ext S,
+    dsimp,
+    simp only [diagram_pullback_app, colimit.ι_pre_assoc,
+      colimit.ι_pre, ι_colim_map_assoc, category.assoc],
+    let e := S.unop.pullback_comp g.unop f.unop,
+    dsimp only [functor.op, pullback_obj],
+    erw [← colimit.w _ e.inv.op, ← category.assoc, ← category.assoc],
+    congr' 1,
+    ext I,
+    dsimp,
+    simp only [multiequalizer.lift_ι, category.assoc],
+    cases I,
+    dsimp only [cover.L.base, cover.L.map],
+    congr' 2,
+    simp,
+  end }
+
+/-- An auxiliary definition used in `plus` below. -/
+@[simps]
+def plus_map {P Q : Cᵒᵖ ⥤ D} (η : P ⟶ Q) : J.plus_obj P ⟶ J.plus_obj Q :=
+{ app := λ X, colim_map
+  { app := λ S, multiequalizer.lift _ _
+      (λ I, multiequalizer.ι (S.unop.index P) I ≫ η.app (op I.Y)) begin
+        intros I,
+        erw [category.assoc, category.assoc, ← η.naturality, ← η.naturality,
+          ← category.assoc, ← category.assoc, multiequalizer.condition],
+        refl,
+      end,
+    naturality' := λ S T e, by { dsimp, ext, simpa } },
+  naturality' := begin
+    intros X Y f,
+    dsimp,
+    ext,
+    simp only [diagram_pullback_app, ι_colim_map, colimit.ι_pre_assoc,
+      colimit.ι_pre, ι_colim_map_assoc, category.assoc],
+    simp_rw ← category.assoc,
+    congr' 1,
+    ext,
+    dsimp,
+    simpa,
+  end }
+
+variable (D)
+
+/-- The plus construction, a functor sending `P` to `J.plus_obj P`. -/
+@[simps]
+def plus : (Cᵒᵖ ⥤ D) ⥤ Cᵒᵖ ⥤ D :=
+{ obj := λ P, J.plus_obj P,
+  map := λ P Q η, J.plus_map η,
+  map_id' := begin
+    intros P,
+    ext,
+    dsimp,
+    simp only [ι_colim_map, category.comp_id],
+    convert category.id_comp _,
+    ext,
+    simp only [multiequalizer.lift_ι, category.id_comp],
+    exact category.comp_id _,
+  end,
+  map_comp' := begin
+    intros P Q R η γ,
+    ext,
+    dsimp,
+    simp only [ι_colim_map, ι_colim_map_assoc],
+    rw ← category.assoc,
+    congr' 1,
+    ext,
+    dsimp,
+    simp,
+  end }
+
+variable {D}
+
+/-- The canonical map from `P` to `J.plus.obj P`.
+See `to_plus` for a functorial version. -/
+@[simps]
+def map_to_plus : P ⟶ (J.plus D).obj P :=
```

Comment:
`map` sounds to much like you are mapping something through a functor or homomorphism. So I thought this might be `to_plus.app`, but dot-notation is also not ideal. At the same time, you want to reserve `to_plus_app` for simps. Would `to_plus_on_obj` be a compromise?
```suggestion
def to_plus_on_obj : P ⟶ (J.plus D).obj P :=
```

Reply:
I agree the name is not ideal. I also realized that there were some @simps generated lemmas like plus_obj_2 because of naming collisions, so I renamed a few things. I'm happy to change the names again if needed.